Sources/Controllers/Panels/OAMapPanelViewController.mm

```diff
@@ -1665,20 +1666,57 @@ - (void) targetPointAddFavorite
     [self targetHideContextPinMarker];
     [self targetHideMenu:.3 backButtonClicked:YES onComplete:nil];
     
-    OAPOI *poi = nil;
-    if ([self.targetMenuView.targetPoint.targetObj isKindOfClass:OAPOI.class])
-        poi = self.targetMenuView.targetPoint.targetObj;
+    NSString *preselectedIconName = [self getIconName:self.targetMenuView.targetPoint.targetObj];
     
     OAEditPointViewController *controller =
             [[OAEditPointViewController alloc] initWithLocation:self.targetMenuView.targetPoint.location
                                                           title:self.targetMenuView.targetPoint.title
                                                     customParam:self.targetMenuView.targetPoint.titleAddress
                                                       pointType:EOAEditPointTypeFavorite
                                                 targetMenuState:nil
-                                                            poi:poi];
```

Comment:
Why is POI removed, why do you pass preselectedIconName here instead of calculating it in PointEditingHandler?